plugins/out_prometheus_remote_write/remote_write.c

```diff
@@ -21,10 +21,40 @@
 #include <fluent-bit/flb_snappy.h>
 #include <fluent-bit/flb_metrics.h>
 #include <fluent-bit/flb_kv.h>
+#include <fluent-bit/flb_signv4.h>
+#include <fluent-bit/flb_aws_credentials.h>
 
 #include "remote_write.h"
 #include "remote_write_conf.h"
 
+#ifdef FLB_HAVE_AWS
+static flb_sds_t add_aws_auth(struct flb_http_client *c,
+                              struct prometheus_remote_write_context *ctx)
+{
+    flb_sds_t signature = NULL;
+    int ret;
+
+    flb_plg_debug(ctx->ins, "Signing request with AWS Sigv4");
+
+    /* AWS Prometheus Sigv4 does not allow the host header to include the port */
+    ret = flb_http_strip_port_from_host(c);
+    if (ret < 0) {
+        flb_plg_error(ctx->ins, "could not strip port from host for sigv4");
+        return NULL;
+    }
+
+    signature = flb_signv4_do(c, FLB_TRUE, FLB_TRUE, time(NULL),
+                              ctx->aws_region, "es",
```

Comment:
Do you want the service here to be "es"? I think that might be a miscopy from the elastic search, no?